include/iterlib/RocksDBIterator.h

```diff
@@ -0,0 +1,65 @@
+//  Copyright (c) 2016, Facebook, Inc.  All rights reserved.
+//  This source code is licensed under the BSD-style license found in the
+//  LICENSE file in the root directory of this source tree. An additional grant
+//  of patent rights can be found in the PATENTS file in the same directory.
+#pragma once
+
+#include <list>
+#include <memory>
+#include <rocksdb/iterator.h>
+
+#include "iterlib/Iterator.h"
+
+namespace iterlib {
+
+inline folly::StringPiece sliceToStringPiece(const rocksdb::Slice s) {
+  return {s.data(), s.size()};
+}
+
+class RocksDBIterator : public Iterator {
+ public:
+  RocksDBIterator(rocksdb::Iterator* iter) : iter_(iter) {}
+
+  const Item& key() const override { return keys_.back(); }
+
+  const Item& value() const override { return values_.back(); }
+
+ protected:
+  bool doNext() override {
+    if (!firstTime_) {
+      iter_->Next();
+    } else {
+      firstTime_ = false;
+    }
+    auto ret = iter_->Valid();
+    if (ret) {
+      keys_.emplace_back(sliceToStringPiece(iter_->key()));
```

Comment:
We also need the Item references returned to be valid. That's a reason for picking `List`instead of `vector`, which invalidates the references on reallocation